storage/src/index/storage.rs

```diff
@@ -282,17 +311,16 @@ impl<T: Translator, V> Index<T, V> {
         let translated_key = self.translator.transform(key);
         let entry = self.map.entry(translated_key);
         match entry {
-            Entry::Occupied(mut occupied_entry) => {
-                let record_ptr = occupied_entry.get_mut();
```

Comment:
I believe this was the main bug with insert... this pointer seems to get (potentially) invalidated when occupied_entry is moved on L289 below.